plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java

```diff
@@ -2549,6 +2549,10 @@ public Optional<TrinoPrincipal> getSchemaOwner(ConnectorSession session, Catalog
     {
         checkState(!isHiveSystemSchema(schemaName.getSchemaName()), "Schema is not accessible: %s", schemaName);
 
+        if (!accessControlMetadata.supportsRoles()) {
```

Comment:
I think it exposes some issue in the engine. If engine controls ownership elsewhere it should not even ask connector for the owner.

Reply:
I think I get it now, it is about using `allow-all` so you now don't return the owner of the schema in such case.

I am not sure if it is the way to go. Support of roles and ownership could be seen independent from each other. I mean returning role as owner would seem odd, but returning an user sounds ok.

Should we move this method implementation to `AccessControlMetadata`?

What about owner for table and view?

Reply:
Moved to `AccessControlMetadata`. `SHOW CREATE` table and view don't show the owner if my understanding is correct.

Reply:
I know, but it is more about the logical ownership. The thing that schema owner is used today should define the actual design decision.